lib/src/native/realm_core.dart

```diff
@@ -1832,15 +1891,15 @@ class RealmLinkHandle {
         classKey = link.target_table;
 }
 
-class RealmResultsHandle extends HandleBase<realm_results> {
+class RealmResultsHandle extends ReleasableHandle<realm_results> {
   RealmResultsHandle._(Pointer<realm_results> pointer) : super(pointer, 872);
 }
 
-class RealmListHandle extends HandleBase<realm_list> {
+class RealmListHandle extends ReleasableHandle<realm_list> {
   RealmListHandle._(Pointer<realm_list> pointer) : super(pointer, 88);
 }
 
-class RealmQueryHandle extends HandleBase<realm_query> {
+class RealmQueryHandle extends ReleasableHandle<realm_query> {
```

Comment:
These changes are not strictly necessary, but without them we can't implement some tests (there are collections that outlive the migration, which crashes the tests when they try to refresh themselves. This will be addressed as part of the handle ownership PR, so I'd rather leave it as is and have better test coverage.